apis/metal3.io/v1alpha1/baremetalhost_validation.go

```diff
@@ -184,3 +244,87 @@ func validateImageURL(imageURL string) error {
 
 	return nil
 }
+
+func checkStatusAnnotation(bmhstatus *BareMetalHostStatus) error {
+
+	operatstatus_allowed := []string{"", "OK", "discovered", "error", "delayed", "detached"}
+	errortype_allowed := []string{"", "provisioned registration error", "registration error", "inspection error", "preparation error", "provisioning error", "power management error"}
+
+	if !stringInSlice(operatstatus_allowed, string(bmhstatus.OperationalStatus)) {
+		return fmt.Errorf("invalid OperationalStatus in StatusAnnotation")
+	}
+
+	if !stringInSlice(errortype_allowed, string(bmhstatus.ErrorType)) {
+		return fmt.Errorf("invalid ErrorType in StatusAnnotation")
+	}
+
+	return nil
+}
+
+func validHwdDetailsAnnotation(hwdetannotaiton string, inspect string) error {
+	if hwdetannotaiton == "" {
+		return nil
+	}
+
+	if inspect != "disabled" {
```

Comment:
@lentzi90 I've a question about the hardwaredetailsannotation, if you check here - https://metal3io.netlify.app/bmo/external_inspection.html , we can add this annotation in case the `HardwareData` is not present. But, since that object is not part of the BMH resource, it is difficult to make that check as part of baremetalhost_validations. 
For now, I've just added the inspection disabled check for now, but that might cause some issues down the line, so I was wondering what would be the reasonable course here - add dependents objects to baremetalhost_validations or move this validation to another file.

Reply:
Good question! I'm not sure what makes sense for this. @furkatgofurov7 @kashifest any ideas how we should handle this?

Reply:
Separate validation check makes sense since it involves multiple objects. If we want to keep it here, we can derive which hwannotaion  belongs to which BMH through owner-reference chain.

Reply:
> @lentzi90 I've a question about the hardwaredetailsannotation, if you check here - https://metal3io.netlify.app/bmo/external_inspection.html , we can add this annotation in case the `HardwareData` is not present. 

I think we are mixing things up here. The external inspection user guide is referring to  `HardwareDetails data in the Status` which do still exist in the BMH. 
HardwareData is completely separate CRD AFAIK which stores `status.hardware` information, essentially the same as in BMH status, but the latter is left in purpose for backwards compatibility and planned to be dropped when there is a new API for BMO

Reply:
Ah true 😀

Reply:
So, as I understand it, the current check if good enough then. Since according to - https://github.com/metal3-io/metal3-docs/blob/main/design/baremetal-operator/external-introspection.md , even if we have hardwaredetails inside the status-annotation, the hardwaredetails-annotation will take precedence.